include/nbl/video/IGPUCommandPool.h

```diff
@@ -88,6 +87,14 @@ class IGPUCommandPool : public core::IReferenceCounted, public IBackendObject
     {
         struct header_t
         {
+        public:
+            header_t(core::LinearAddressAllocator<uint32_t>&& alloc):
+                commandAllocator(std::move(alloc)),
+                next(nullptr),
+                nextHead(nullptr),
+                prevHead(nullptr)
+            {}
+            
             core::LinearAddressAllocator<uint32_t> commandAllocator;
```

Comment:
dont do this,  do perfect forwarding instead
```cpp
template<typename... Args>
inline header_t(Args&&... args) : commandAllocator(std::forward<Args>(args)...) {}
```
the other members you need not do anything for, they'll initialize to their inline default values